share/functions/prompt_pwd.fish

```diff
@@ -8,18 +8,34 @@ function prompt_pwd --description "Print the current working directory, shortene
         return 0
     end
 
-    # This allows overriding fish_prompt_pwd_dir_length from the outside (global or universal) without leaking it
+    # This allows overriding fish_prompt_pwd_dir_length and fish_prompt_pwd_depth from the outside (global or universal) without leaking it
     set -q fish_prompt_pwd_dir_length
     or set -l fish_prompt_pwd_dir_length 1
+  
+    set -q fish_prompt_pwd_depth
+    or set -l fish_prompt_pwd_depth 0
 
     # Replace $HOME with "~"
     set realhome ~
     set -l tmp (string replace -r '^'"$realhome"'($|/)' '~$1' $PWD)
+    # Split $temp to an array(for the trim depth only) 
+    set -l tempArr (string split '/' $tmp)
 
     if [ $fish_prompt_pwd_dir_length -eq 0 ]
-        echo $tmp
+
+        if [ $fish_prompt_pwd_depth -gt (count $tempArr) -o $fish_prompt_pwd_depth -eq 0 ] #dont trim, dont shorten
```

Comment:
If you don't check for integerness above, this can throw an error.

Both in cases like "banana", but also when it has multiple elements.